src/Xamarin.Android.Build.Tasks/Tasks/ResolveSdksTask.cs

```diff
@@ -302,6 +306,69 @@ public bool RunTask ()
 			return !Log.HasLoggedErrors;
 		}
 
+		static readonly Regex javaVersionRegex = new Regex (@"""([\d\.]+)_\d+""");
+
+		Version GetJavaVersionForFramework (string targetFrameworkVersion)
+		{
+			var apiLevel = AndroidVersion.TryOSVersionToApiLevel (targetFrameworkVersion);
+			if (apiLevel >= 24)
+				return new Version (1, 8);
+			else if (apiLevel == 23)
+				return new Version (1, 7);
+			else
+				return new Version (1, 6);
+		}
+
+		Version GetJavaVersionForBuildTools (string buildToolsVersion)
+		{
+			Version buildTools;
+			if (!Version.TryParse (buildToolsVersion, out buildTools)) {
+				return new Version (1, 7);
+			}
+			if (buildTools >= new Version (24, 0, 1))
+				return new Version (1, 8);
+			return new Version (1, 7);
+		}
+
+		void ValidateJavaVersion (string targetFrameworkVersion, string buildToolsVersion)
+		{
+			Version requiredJavaForFrameworkVersion = GetJavaVersionForFramework (targetFrameworkVersion);
+			Version requiredJavaForBuildTools = GetJavaVersionForBuildTools (buildToolsVersion);
+
+			Version required = requiredJavaForFrameworkVersion > requiredJavaForBuildTools ? requiredJavaForFrameworkVersion : requiredJavaForBuildTools;
+			
+			var proc = new Process ();
+			proc.StartInfo.FileName = Path.Combine (JavaSdkPath, "bin", OS.IsWindows ? "java.exe" : "java");
```

Comment:
Perhaps this should be using `MonoAndroidHelper.RunProcess()`? That should be a bit simpler.